crates/iceberg-catalog/src/service/storage/mod.rs

```diff
@@ -142,9 +162,23 @@ impl StorageProfile {
                         table_id,
                         namespace_id,
                         data_access,
-                        secret.map(|s| match s {
-                            StorageCredential::S3(s) => s,
-                        }),
+                        secret.map(|s| s.try_to_s3()).transpose()?,
+                    )
+                    .await
+            }
+            StorageProfile::Azdls(profile) => {
+                profile
+                    .generate_table_config(
+                        warehouse_id,
+                        table_id,
+                        namespace_id,
+                        data_access,
+                        table_location,
+                        secret
+                            .ok_or_else(|| {
+                                CredentialsError::MissingCredential(self.storage_type())
```

Comment:
By doing this we rule out the possibility of a credential-free adls - completely public. I am not sure if this is relevant- would the storage driver support it?

Reply:
Reqsign blows up if there's no credential, even worse, it'll take whatever it can find in env if nothing is passed to it which may cause really weird behavior if some az creds end up mounted into a catalog container, something like a profile being created for a warehouse based on the catalog provider's az account.